src/net/gateway-web3.ts

```diff
@@ -190,69 +195,129 @@ export class Web3Gateway {
+        ]
+        if (ens) metrics.push(this.checkEns())
+        return Promise.all(metrics).then(() => {
+                // ok
+                // maybe flag it as already checked
+            })
     }
 
     /** Determines whether the current Web3 provider is syncing blocks or not. Several types of prviders may always return false. */
-    public isSyncing(): Promise<boolean> {
-        if (!this._provider) return Promise.resolve(false)
+    public isSyncing(): Promise<void> {
+        if (!this._provider) return Promise.reject()
         else if (this._provider instanceof JsonRpcProvider || this._provider instanceof Web3Provider || this._provider instanceof IpcProvider || this._provider instanceof InfuraProvider) {
-            return this._provider.send("eth_syncing", []).then(result => !!result)
+            let performanceTime = new Date().getTime()
+            return this._provider.send("eth_syncing", []).then(result => {
+                performanceTime = Math.round(new Date().getTime() - performanceTime)
+                this.performanceTime = performanceTime > this.performanceTime ? performanceTime : this.performanceTime
+                return !!result ? Promise.reject() : Promise.resolve()
+            })
         }
-        // else if (this._provider instanceof FallbackProvider || this._provider instanceof EtherscanProvider) {}
 
-        return Promise.resolve(false)
+        return Promise.reject()
     }
 
     /** Request the amount of peers the Gateway is currently connected to */
-    public getPeers(): Promise<number> {
-        if (!this._provider) return Promise.resolve(0)
+    public getPeers(): Promise<void> {
+        // TODO Review if not rejecting and setting peerCount = -1 is the best solution
+        if (!this._provider) return Promise.reject()
         else if (!(this._provider instanceof JsonRpcProvider) && !(this._provider instanceof Web3Provider) &&
             !(this._provider instanceof IpcProvider) && !(this._provider instanceof InfuraProvider)) {
-            return Promise.resolve(0)
+            return Promise.reject()
         }
 
+        let performanceTime = new Date().getTime()
         return this._provider.send("net_peerCount", [])
             .then(result => {
+                // TODO maybe not needed Exception here
                 if (!result) throw new Error('peersCount not available for web3 gateway')
-                return BigNumber.from(result).toNumber()
+                performanceTime = Math.round(new Date().getTime() - performanceTime)
+                this.performanceTime = performanceTime > this.performanceTime ? performanceTime : this.performanceTime
+                this.peerCount = BigNumber.from(result).toNumber()
             })
             .catch(err => {
-                return 0
+                this.peerCount = -1
+            })
+    }
+
+    /** Request the block number of the Gateway which is currently connected to */
+    public getBlockNumber(): Promise<void> {
```

Comment:
Why do we need a wrapper around `getBlockNumber`?

Do we simply want it to do the performance measurements?

Reply:
There are two reasons for that. The first one is what you comment, the performance measurements, and the other one is to have some kind of "standard" inside the Web3Gateway, where every metric is a `Promise<void>` and the value of the metric (the block number in this case) is stored in a variable in the Object.